src/ngraph/runtime/intelgpu/intelgpu_backend.cpp

```diff
@@ -279,6 +280,44 @@ bool runtime::intelgpu::IntelGPUBackend::compile(shared_ptr<Function> func)
                                               output_size);
             topology.add(cldd_pooling);
         }
+        else if ("AvgPool" == op->description())
+        {
+            arguments_check(op, 1, 1);
+
+            const string& input_name = op->get_inputs().begin()->get_tensor().get_name();
+            const string& output_name = op->get_outputs().begin()->get_tensor().get_name();
+            const Shape& out_shape = op->get_outputs().begin()->get_shape();
+            const cldnn::tensor output_size =
+                runtime::intelgpu::IntelGPULayout::create_cldnn_tensor(out_shape);
+
+            const shared_ptr<op::AvgPool> avg_pool = static_pointer_cast<op::AvgPool>(op);
+
+            const Shape& pool_shape = avg_pool->get_window_shape();
+            const Strides& pool_strides = avg_pool->get_window_movement_strides();
+            const Shape& pad_below = avg_pool->get_padding_below();
+            const Shape& pad_above = avg_pool->get_padding_above();
+
+            vector<cldnn::tensor::value_type> offset({0, 0, 0, 0});
+            size_t ridx = 4;
+            for (auto i = pad_below.crbegin(); i != pad_below.crend() && ridx > 0; ++i, --ridx)
+            {
+                offset.at(ridx - 1) = -(*i);
+            }
+
+            const cldnn::tensor input_offset(
```

Comment:
To avoid this code duplication, could you please look into `IntelGPULayout::create_cldnn_tensor()` function and create `IntelGPULayout::create_cldnn_offset()` with same approach? 
After this, please use the function in MaxPool and AvgPool operations.